databricks/koalas/groupby.py

```diff
@@ -1594,17 +1594,36 @@ def head(self, n=5):
         10    10
         Name: b, dtype: int64
         """
-        sdf = self._kdf._sdf
+        if isinstance(self, SeriesGroupBy):
+            kdf = self._kser._kdf
+        else:
+            kdf = self._kdf
+
+        if self._agg_columns_selected:
+            agg_columns = self._agg_columns
+        else:
+            agg_columns = [kdf._kser_for(label) for label in kdf._internal.column_labels]
+
+        groupkey_labels = [
+            verify_temp_column_name(kdf, "__groupkey_{}__".format(i))
+            for i in range(len(self._groupkeys))
+        ]
+        kdf = kdf[
+            [s.rename(label) for s, label in zip(self._groupkeys, groupkey_labels)] + agg_columns
+        ]
+        groupkey_scols = [kdf._internal.spark_column_for(label) for label in groupkey_labels]
```

Comment:
@ueshin, it looks good but can we have a private function for these cases? Looks like it's duplicated in https://github.com/databricks/koalas/pull/1472 https://github.com/databricks/koalas/pull/1473/files and https://github.com/databricks/koalas/pull/1471.

Reply:
Sure, I'll address it in the following PRs.